backend/GameManager.py

```diff
@@ -28,20 +52,20 @@ def placement_complete(self):
         return self.pieces.all_pieces_placed()
 
     def place_piece(self, row, column):
-        if (row, column) in self.current_moves:
-            if self.board.check_position(row, column) != CellType.EMPTY:
+        if (row, column) in self.open_moves:
+            if self.board.check_position(row, column) != Color.EMPTY:
                 print(self.board.check_position(row, column))
                 return "GameManagerError -- position not empty"
 
             is_piece_placed = 0
-            if self.turn == Turn.WHITE:
+            if self.turn == Color.WHITE:
                 is_piece_placed = self.pieces.set_white_piece(row, column)
             else:
                 is_piece_placed = self.pieces.set_black_piece(row, column)
 
             if is_piece_placed == 1:
                 self.board.set_position(row, column, self.turn.name.lower())
-                self.current_moves.remove((row, column))
+                self.open_moves.remove((row, column))
                 return 1
             # else error -- figure out handling. are we making error classes?
```

Comment:
It wont let me comment on the `set_white_piece` function but right now its just returning a string and it needs to raise an exception then we can add a try-catch here for that to handle this case.

Reply:
this was there before i start but i agree there should be something we need to do about this